### PR TITLE
Change Prep init to pass folders as parameter

### PR DESCRIPTION
**Pull request by linhandev**
Pass folders to Prep with init parameters, this way the prep class couples less with subclass

### CODE REVIEW

File: tools/prepare.py [hunk 1]
```diff
@@ -39,16 +41,46 @@
...
+    def __init__(
+        self,
+        dataset_fdr,
+        urls,
+        image_fdr,
+        label_fdr,
+        phase_fdr="phase0",
+        raw_fdr="raw",
+        datasets_root="data",
+        visualize_fdr="vis",
```
Comment (shiyutang) [inline tools/prepare.py:53]: visualize file can be stored in each experiment path

File: tools/prepare.py [hunk 2]
```diff
@@ -39,16 +41,46 @@
...
-        self.raw_data_path = None
-        self.image_dir = None
-        self.label_dir = None
-        self.urls = None
+    def __init__(
+        self,
+        dataset_fdr,
+        urls,
+        image_fdr,
+        label_fdr,
```
Comment (shiyutang) [inline tools/prepare.py:49]: These attributes can have default params.

**Review by shiyutang — CHANGES_REQUESTED** [review]

File: tools/prepare.py (hunk 2 above)
Comment (linhandev) [reply tools/prepare.py:49]: This directory varies from dataset to dataset. 

These two params will be used in `self.image_dir`, this path is the raw path for the dataset / image folder in zip. Like for msd task 1 `self.image_fdr` is Task01_BrainTumour/Task01_BrainTumour/imagesTr

File: tools/prepare.py (hunk 1 above)
Comment (linhandev) [reply tools/prepare.py:53]: Roger.

I planed to store visualization for preprocessed data here but it isnt really used.

Will look into visualization more thoroughly this week. Maybe we can just merge it like this for now. I can fix this in a future pr. This way I won't need to change all the latter prs.

File: tools/prepare.py (hunk 2 above)
Comment (shiyutang) [reply tools/prepare.py:49]: I see

File: tools/prepare.py (hunk 1 above)
Comment (shiyutang) [reply tools/prepare.py:53]: ok

**Review by shiyutang — COMMENTED** [review]
You may want to run the prepare_lung_coronavirus.py to see if the result directory is as follows. Just in case something went wrong.
image


If the result is satisfactory, then this PR is ready to merge.

**Comment by shiyutang** [comment]
The content in this PR have been incorporate into the current code

**Pull request closed without merging**